implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java

```diff
@@ -69,7 +71,13 @@
     @ConfigProperty(name = "io.smallrye.health.emptyChecksOutcome", defaultValue = "UP")
     String emptyChecksOutcome;
 
-    private List<HealthCheck> additionalChecks = new ArrayList<>();
+    @Inject
+    @Liveness
+    LivenessHealthRegistry livenessHealthRegistry;
```

Comment:
Why don't you declare an `HealthRegistry` and have the type safe injection discrimination done with the qualifier?

Reply:
the `getChecks()` method is only available in our implementation, not in the interface. Would it make sense to have it also in the interface?

Reply:
+1 to add the `getCheck()` to the interface

Reply:
@antoinesd I've decided to remove this method from the interface. This is too restrictive for the implementation. I've run into this problem myself when I changed underlying collection in our implementations of health registry to `Map<String, Uni<HealthCheckResponse>>` as our impl now tracks all checks in Unis.